identity/ipa/ipa_host.py

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ipa_host
+short_description: Manage FreeIPA host
+description:
+- Add, modify and delete an IPA host using IPA API
+options:
+  fqdn:
+    description:
+    - Full qualified domain name.
+    - Can not be changed as it is the unique identifier.
+    required: true
+    aliases: ["name"]
+  description:
+    description:
+    - A description of this host.
+    required: false
+  force:
+    description:
+    - Force host name even if not in DNS.
+    required: false
+  ip_address:
+    description:
+    - Add the host to DNS with this IP address.
+    required: false
+  macddress:
```

Comment:
typo here.

As a side note, IMHO the naming of the args should be more consistent, like `macaddress` vs `ip_address`. I would assume, the arg is named like `mac_address`. Also later for `nshostlocation` it makes things more readable in snake case `ns_host_location`. But up to your choice.